tests/PHPUnit/System/ArchiveInvalidationTest.php

```diff
@@ -201,7 +201,18 @@ public function testDisablePluginArchiveBySiteId()
         $this->assertFalse(Rules::isSegmentPluginArchivingDisabled('testPlugin', 1));
     }
 
+    /**
+     * Check that both date range types are correctly parsed without any exceptions
+     */
+    public function testDateRangesCorrectlyParsed()
+    {
+        $testRanges = ['2020-01-01,2020-03-31', 'last30'];
 
+        foreach ($testRanges as $dateRange) {
+            $r = new Request("module=API&method=CoreAdminHome.invalidateArchivedReports&idSites=".self::$fixture->idSite1."&period=range&dates=".$dateRange);
```

Comment:
I guess it would be useful to also have some tests that check the return values of `getDatesToInvalidateFromString`.
You could directly call that private method with code like
```
    $api = API::getInstance();
    $reflection = new \ReflectionClass(API::class);
    $method = $reflection->getMethod('getDatesToInvalidateFromString');
    $method->setAccessible(true);

    self::assertEquals([], $method->invokeArgs(API::getInstance(), $parameters));
```

Reply:
Thanks for the example code :+1: I've added additional tests to directly check the test ranges against the `getDatesToInvalidateFromString()` method result.